pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3IndexFilterOperator.java

```diff
@@ -69,30 +76,16 @@ public H3IndexFilterOperator(Predicate predicate, IndexSegment indexSegment, int
     } else {
       throw new RuntimeException("Expecting one of the arguments of ST_DISTANCE to be an identifier");
```

Comment:
better to add this logic in filter plan node and fallback to expressionbased filter operator

Reply:
Yes, filter plan node has this logic, so this is defensive checking.